spec/lookup_spec.rb

```diff
@@ -32,24 +30,28 @@
           :key_conditions=>{
             "component_key"=> {
               :comparison_operator=>"EQ",
-              :attribute_value_list=>[{"s"=>"id/9dd916afd5516828a91d259967fd394a"}]
+              :attribute_value_list=>[{"s"=>"id/218c835cec343537589dbf1619532e4d"}]
             },
             "batch_version"=>{
               :comparison_operator=>"EQ",
-              :attribute_value_list=>[{"n"=>"{:variant=>\"foo\"}"}]
+              :attribute_value_list=>[{"n"=>"0"}]
             }
           }
         }
       end
 
-      it "queries DynamoDb and returns a location" do
-        AWS::DynamoDB::Client::V20120810
-          .any_instance
-          .stub(:initialize)
+      it "queries DynamoDb and returns a location when not in cache" do
+        expect_any_instance_of(Dalli::ElastiCache).to receive(:initialize)
+        expect_any_instance_of(Dalli::ElastiCache).to receive(:client).and_return(Dalli::Client.new)
```

Comment:
Is it best to return a new object (which would be the object the class would return), or a double? I kept changing it, and decided to leave it as this for now.

Also the `expect_any_instance_of` somehow leaks out into the next test! I'll wait for feedback on the above and revisit if it's needed. I took it out of the next test for now, so they all pass. :wink: